homeassistant/components/owntracks/__init__.py

```diff
@@ -180,7 +180,34 @@
             return json_response([])
 
     hass.helpers.dispatcher.async_dispatcher_send(DOMAIN, hass, context, message)
-    return json_response([])
+
+    response = []
+
+    persons = list(filter(lambda x: x.domain == "person", hass.states.async_all()))
+
+    for person in persons:
+        if "latitude" in person.attributes and "longitude" in person.attributes:
+            response.append(
+                {
+                    "_type": "location",
+                    "lat": person.attributes["latitude"],
+                    "lon": person.attributes["longitude"],
+                    "tid": person.attributes["id"][-2:],
```

Comment:
why this?

Reply:
The "tid" in Owntracks is just 2 characters, and is generated in a similar way - the last 2 characters of the device's name - https://owntracks.org/booklet/features/tid/#tid

Reply:
> The TID of your friends is displayed on the OwnTracks map if you don't have an address book entry associated with that friend, and if said friend has no retained card published.

Better to generate one based on the name? `''.join(p[0] for p in person.name.split(' '))`

Reply:
If the person's name's "John" that'd generate `tid` of "John", which is more than 2 characters, or am I wrong in interpreting the code?

My point being this might generate unexpected behavior on the owntracks' client side.

Reply:
Did you try the code?

```text
>>> name = "John"
>>> ''.join(p[0] for p in name.split(' '))
'J'
```

Reply:
To limit it to 2 chars: `''.join(p[0] for p in name.split(' ')[:2])`